### PR TITLE
JIT: Mark constant arguments of HWINTRINSIC nodes as DONT_CSE

### LINKED ISSUE #63432: Poor codegen for Vector128.Create() with constants sharing the same value
### Description
Calling `Vector128.Create(float, float, float, float)` with constants that share the same value results in poor codegen, as demonstrated below.

[Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho1iAZiakGAYQYBvGg11NySBgDUYYDNHKkAHAB4AZgBsI2DAD4GAWXIAKAJRadPSDiAHZjU3MoSysWNVgXGG9yOzRSFJR0u185aiCAX0DdZkMTMwtreycXdw9SPwDcoL1Q8LKo61j4jETk1JTFTOzChgLqPKA)
```cs
using System;
using System.Runtime.Intrinsics;

public class C {
    static Vector128<float> M1() {
        return Vector128.Create(1f,2f,4f,4f);
    }
    static Vector128<float> M2() {
        return Vector128.Create(1f,2f,3f,4f);
    }
}
```
Output assembly (CoreCLR 6.0.21.52210 on amd64)

```asm
C.M1()
    L0000: vzeroupper
    L0003: vmovss xmm0, [0x7ffb22bb0480]
    L000b: vmovss xmm1, [0x7ffb22bb0484]
    L0013: vinsertps xmm0, xmm0, xmm1, 0x10
    L0019: vmovss xmm1, [0x7ffb22bb0488]
    L0021: vmovaps xmm2, xmm1
    L0025: vinsertps xmm0, xmm0, xmm2, 0x20
    L002b: vinsertps xmm0, xmm0, xmm1, 0x30
    L0031: vmovupd [rcx], xmm0
    L0035: mov rax, rcx
    L0038: ret

C.M2()
    L0000: vzeroupper
    L0003: vmovupd xmm0, [0x7ffb22bb04c0]
    L000b: vmovupd [rcx], xmm0
    L000f: mov rax, rcx
    L0012: ret
```

This seems to affect all other `VectorXXX.Create()` functions, but only for `float` and `double`.

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes: https://github.com/dotnet/runtime/issues/63432

another case where it helps:
```csharp
static void Test()
{
    for (int i = 0; i < 10; i++)
    {
        Console.WriteLine(
            Sse.MoveMask(Vector128.Create(1f, 2f, 3f, 4f)));
    }
}
```
codegen diff: https://www.diffchecker.com/lyEPUSdN

### CODE REVIEW

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -14152,6 +14152,17 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
...
         *use = fgMorphTree(*use);
         multiOp->gtFlags |= ((*use)->gtFlags & GTF_ALL_EFFECT);
+
+#if defined(FEATURE_HW_INTRINSICS)
+        if (opts.OptimizationEnabled() && multiOp->OperIs(GT_HWINTRINSIC) && (*use)->OperIsConst())
+        {
+            // Never do CSE for constant arguments of HWINTRINSICS node such as
+            // MoveMask(vec, 0x...) or Vector.Create(1f,2f,3f,3f)
+            // The only case that might suffer from this is `Vector.Create(1f, x, 3f, 3f)` but this specific
+            // one is rare and needs a better treatment anyway (e.g. loading '1,0,3,3` from .data and then do insert)
```
Comment (EgorBo) [inline src/coreclr/jit/morph.cpp:14162]: and it does need a better treatment, consider this: https://sharplab.io/#v2:C4LghgzgtgPgAgBgARwIwDoBKBXAdsASygFN0BJfAJwNwgIGMIBuAWACh32A1Y+4Ae0oAmAKwA2ADwAzADb8wwAHxIAKsQjAAFLPnAkADwCUSALzL2SS0h59BosegDClYguKbUUgDRIh3pADM/vo+Iv5i/gDswYZMQA=

File: src/coreclr/jit/morph.cpp [hunk 2]
```diff
@@ -14185,6 +14186,39 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
                 }
                 break;
             }
+#endif
+
+            case NI_Vector128_Create:
```
Comment (tannergooding) [inline src/coreclr/jit/morph.cpp:14191]: We should likely add back the handling (or do it in a separate PR) to have the same for any intrinsic that is `HWIntrinsicInfo::lookupCategory(intrinsicId) == HW_Category_IMM` (x86/x64) or `HWIntrinsicInfo::HasImmediateOperand(intrinsicId)` (ARM64)

Comment (tannergooding) [reply src/coreclr/jit/morph.cpp:14191]: In those cases, the instruction has one or more non-SIMD operands (typically a `byte imm`) that is functionally "required" to be a constant, as otherwise it will risk expanding to a very expensive jump table.

Comment (EgorBo) [reply src/coreclr/jit/morph.cpp:14217]: @tannergooding could you please look one more time? I've addressed your feedback and fixed regressions - they were coming from CreateScalarUnsafe, e.g. 
```csharp
public float Test()
{
    return Vector128.CreateScalarUnsafe(5.0f).ToScalar()
           + Vector128.CreateScalarUnsafe(5.0f).ToScalar();
}
```

**Review by tannergooding — APPROVED** [review]
